pkg/registry/batch/job/storage/storage.go

```diff
@@ -101,3 +115,105 @@ func (r *StatusREST) Get(ctx genericapirequest.Context, name string, options *me
+	case extensionsv1beta1.SchemeGroupVersion:
+		return extensionsv1beta1.SchemeGroupVersion.WithKind("Scale")
+	case appsv1beta1.SchemeGroupVersion:
+		return appsv1beta1.SchemeGroupVersion.WithKind("Scale")
+	case appsv1beta2.SchemeGroupVersion:
+		return appsv1beta2.SchemeGroupVersion.WithKind("Scale")
+	default:
+		return autoscalingv1.SchemeGroupVersion.WithKind("Scale")
+	}
+}
+
+// New creates a new Scale object
+func (r *ScaleREST) New() runtime.Object {
+	return &autoscaling.Scale{}
+}
+
+func (r *ScaleREST) Get(ctx genericapirequest.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	job, err := r.registry.GetJob(ctx, name, options)
+	if err != nil {
+		return nil, errors.NewNotFound(extensions.Resource("jobs/scale"), name)
+	}
+	scale, err := scaleFromJob(job)
+	if err != nil {
+		return nil, errors.NewBadRequest(fmt.Sprintf("%v", err))
+	}
+	return scale, nil
+}
+
+func (r *ScaleREST) Update(ctx genericapirequest.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc) (runtime.Object, bool, error) {
+	job, err := r.registry.GetJob(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, false, errors.NewNotFound(extensions.Resource("job/scale"), name)
+	}
+
+	oldScale, err := scaleFromJob(job)
+	if err != nil {
+		return nil, false, err
+	}
+
+	obj, err := objInfo.UpdatedObject(ctx, oldScale)
+	if err != nil {
+		return nil, false, err
+	}
+	if obj == nil {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("nil update passed to Scale"))
+	}
+	scale, ok := obj.(*autoscaling.Scale)
+	if !ok {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("expected input object type to be Scale, but %T", obj))
+	}
+
+	if errs := autoscalingvalidation.ValidateScale(scale); len(errs) > 0 {
+		return nil, false, errors.NewInvalid(extensions.Kind("Scale"), name, errs)
+	}
+
+	job.Spec.Parallelism = &scale.Spec.Replicas
+	job, err = r.registry.UpdateJob(ctx, job, createValidation, updateValidation)
+	if err != nil {
+		return nil, false, err
+	}
+	newScale, err := scaleFromJob(job)
+	if err != nil {
+		return nil, false, errors.NewBadRequest(fmt.Sprintf("%v", err))
+	}
+	return newScale, false, nil
+}
+
+// scaleFromJob returns a scale subresource for a job.
+func scaleFromJob(job *batch.Job) (*autoscaling.Scale, error) {
+	selector, err := metav1.LabelSelectorAsSelector(job.Spec.Selector)
+	if err != nil {
+		return nil, err
+	}
+	return &autoscaling.Scale{
+		// TODO: Create a variant of ObjectMeta type that only contains the fields below.
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              job.Name,
+			Namespace:         job.Namespace,
+			UID:               job.UID,
+			ResourceVersion:   job.ResourceVersion,
+			CreationTimestamp: job.CreationTimestamp,
+		},
+		Spec: autoscaling.ScaleSpec{
+			Replicas: *job.Spec.Parallelism,
+		},
+		Status: autoscaling.ScaleStatus{
+			Replicas: *job.Spec.Parallelism,
```

Comment:
I expected status to come from status.  typo?

Reply:
No, there's no parallelism in status.

Reply:
That seems weird.  Definitely worth a comment.

Reply:
it's also unclear how status.replicas should/would be affected by job.spec.completions

Reply:
We could use `status.Active`, which will give you information about currently running instances of a job. But, it can happen that it will be less than specified parallelism! This will happen when `spec.Parallelism > 1` and `spec.Completions - status.Succeeded < spec.Parallelism`.
Of course this means your job is close to finish, and the provided information will be accurate but I'm worried the client will be trying to scale up, since it didn't get the desired `replicas`. 
One other option is to introduce parallelism in status, to be able to notify scale client that the desired parallelism was properly picked by the controller.

Reply:
There's one more option that just popped in my head, which seems the most reasonable. To allow scaling job only when there's room, iow. `spec.Completions - status.Succeeded < spec.Replicas`, in all other situations scale would fail. @liggitt wdyt?